node/sc/vt_recovery.go

```diff
@@ -281,15 +281,19 @@ pendingTxLoop:
 				logger.Trace("filtered pending nonce", "requestNonce", it.Event.RequestNonce, "handledNonce", hint.handleNonce)
 				pendingEvents = append(pendingEvents, &RequestValueTransferEvent{it.Event})
 				if len(pendingEvents) >= maxPendingTxs {
+					it.Close()
 					break pendingTxLoop
 				}
 			}
 		}
 		startBlkNum = endBlkNum + 1
 		endBlkNum = startBlkNum + filterLogsStride
+		it.Close()
```

Comment:
How about `defer it.Close()`?

Reply:
This is in the for loop. So `defer` is not good way.

Reply:
We can use anonymous func() as delegate, but it is okay :)